client/verta/verta/dataset/_dataset.py

```diff
@@ -93,14 +93,106 @@ def _set_commit_and_blob_path(self, commit, blob_path):
         self._commit = commit
         self._blob_path = blob_path
 
+    def _get_components_to_download(self, component_path, download_to_path=None):
+        """
+        Identify components to be downloaded, along with their local destination paths.
+
+        Parameters
+        ----------
+        component_path : str
+            Path to directory or file within blob.
+        download_to_path : str, optional
+            Local path to download to.
+
+        Returns
+        -------
+        components_to_download : dict
+            Map of component paths to local destination paths.
+        downloaded_to_path : str
+            Path where file(s) were downloaded to. May be relative to cwd.
+
+        """
+        implicit_download_to_path = download_to_path is None
+
+        component_path_as_dir = component_path if component_path.endswith('/') else component_path+'/'
+        components_to_download = dict()
+
+        for component_blob in self._path_component_blobs:
+            if component_blob.path == component_path:  # exact match with file
+                if implicit_download_to_path:
+                    # default to filename from `component_path` in cwd
+                    local_path = os.path.basename(component_path)
+
+                    # avoid collision with existing file
+                    while os.path.exists(local_path):
+                        local_path = _file_utils.increment_path(local_path)
+                else:
+                    local_path = download_to_path
+
+                return ({component_blob.path: local_path}, local_path)
+            elif component_blob.path.startswith(component_path_as_dir):
+                if implicit_download_to_path:
+                    # default to path relative to parent of `component_path`
+                    #     For example:
+                    #     component_blob.path = "coworker/downloads/data/info.csv"
+                    #     component_path      = "coworker/downloads"
+                    #     local_path          =          "downloads/data/info.csv"
+                    local_path = os.path.relpath(
```

Comment:
Can we call this something else and add to your example above? I think it would be `data/info.csv`, right?

Reply:
It is actually already `downloads/data/info.csv`. The reassignment later is in case `downloads/` already exists, in which case `local_path` becomes `downloads 1/data/info.csv`. I'll try to clarify that in the comments.